components/NavigationDrawer/src/MDCBottomDrawerViewController.m

```diff
@@ -249,6 +256,64 @@ - (void)bottomDrawerTopTransitionRatio:
   }
 }
 
+- (void)bottomDrawerPresentTransitionDidEnd:
+    (MDCBottomDrawerPresentationController *)presentationController {
+  _currentTransitionCoordinator = nil;
+  if ([self.delegate respondsToSelector:@selector(bottomDrawerControllerDidEndOpenTransition:)]) {
+    [self.delegate bottomDrawerControllerDidEndOpenTransition:self];
+  }
+}
+
+- (void)bottomDrawerDismissTransitionDidEnd:
+    (MDCBottomDrawerPresentationController *)presentationController {
+  _isDrawerClosed = YES;
+  _currentTransitionCoordinator = nil;
+  if ([self.delegate respondsToSelector:@selector(bottomDrawerControllerDidEndCloseTransition:)]) {
+    [self.delegate bottomDrawerControllerDidEndCloseTransition:self];
+  }
+}
+
+- (void)bottomDrawerPresentTransitionWillBegin:
+            (MDCBottomDrawerPresentationController *)presentationController
+                               withCoordinator:
+                                   (id<UIViewControllerTransitionCoordinator>)transitionCoordinator
+                                 targetYOffset:(CGFloat)targetYOffset {
+  _isDrawerClosed = NO;
+  _currentTransitionCoordinator = transitionCoordinator;
+  _lastOffset = targetYOffset;
+  if ([self.delegate respondsToSelector:@selector
+                     (bottomDrawerControllerWillTransitionOpen:withCoordinator:targetYOffset:)]) {
+    [self.delegate bottomDrawerControllerWillTransitionOpen:self
+                                            withCoordinator:transitionCoordinator
+                                              targetYOffset:targetYOffset];
+  }
+}
+
+- (void)bottomDrawerDismissTransitionWillBegin:
+            (MDCBottomDrawerPresentationController *)presentationController
+                               withCoordinator:(id<UIViewControllerTransitionCoordinator>)
+                                                   transitionCoordinator {
+  _currentTransitionCoordinator = transitionCoordinator;
+  if ([self.delegate respondsToSelector:@selector(bottomDrawerControllerWillTransitionClosed:
+                                                                             withCoordinator:)]) {
+    [self.delegate bottomDrawerControllerWillTransitionClosed:self
+                                              withCoordinator:transitionCoordinator];
+  }
+}
+
+- (void)bottomDrawerTopDidChangeYOffset:
+            (MDCBottomDrawerPresentationController *)presentationController
+                                yOffset:(CGFloat)yOffset {
+  // Only forward changes along if the drawer is actually still on screen and the offset has
+  // changed.
+  if (!_isDrawerClosed && _lastOffset != yOffset &&
```

Comment:
Yes I added that to avoid the bouncing the UI after the drawer is closed. The way the offset change is triggered is by the scrollview of the container being updated, which it gets updated to a zero offset for some reason after the drawer is closed. It also gets updated to the zero offset a few times before the drawer opens. I wasn't able to easily change that behavior so I decided to catch it here and prevent the client from having to deal with it. 

As for the lastOffset check -- that is just to avoid sending through duplicate offset changes so the client only gets actual changes and doesn't have to de-dup.

Reply:
updateViewWithContentOffset is called alot and doesn't dedup. There are also interim calls that set to zero as prep then a real value so this will avoid passing those thru to the client.